### ISSUE TITLE
Feature Request: Ability to append directory to TARGET_SERVER

### ISSUE THREAD

**Issue by bkraul** [guysoft/nginx-proxy-pass-docker#1]
**Case Scenario**
I am trying to proxy requests to my Plex server. Generally, Plex exposes its web UI under the web directory, meaning the TARGET_SERVER would have to be [Server IP]:32400/web. But when using such value, the proxy does not honor the subdir, and it reverts to the root dir, which means I have to append /web to my requests.

**Comment by guysoft** [comment]
It would require changing this path: https://github.com/guysoft/nginx-proxy-pass-docker/blob/master/default.conf#L5
I am not sure how to change

**Comment by bkraul** [comment]
I don't have much experience with nginx location syntax, but I'm going to look into this. I believe I read somewhere where someone was trying to do this with jwilder/nginx-proxy.

**Comment by guysoft** [comment]
Ok,  check if this solves it, its a tag ``guysoft/nginx-proxy-pass:target-server`` on docker hub, see commit above, its untested at all!

**Comment by bkraul** [comment]
Will do. Tag doesn't seem to be on hub yet, but will check a bit later.

**Comment by guysoft** [comment]
Its queued to build,  give it like 10 mins. You can check on docker hub

**Comment by bkraul** [comment]
OK, I was able to find the following 2 issues:

1. The ```HTTP_PATH``` variable should be applied on the proxy_pass line of the ```default.conf``` file, like so:
```nginx
proxy_pass              <proxy_pass_protocol_placeholder>://<proxy_pass_placeholder><target_path_placeholder>;
```
This means that location needs to remain ```/``` because we are trying to make the root of the target redirect to the subdir in the upstream.

 2. There is an issue with the sed substitution in this line of the commit on ```start.sh```.

```bash
/bin/sed -i "s/<target_path_placeholder>/${HTTP_PATH}/" /etc/nginx/conf.d/default.conf
````
You probably need to somehow escape the ```/``` when it is entered. I am setting the ENV variable like so:
```- HTTP_PATH=/web```.

I believe sed gets pretty picky with reserved characters (```/``` is one of them), so you'd have to use something like ```\/``` (through a replacement call before calling ```sed```).

I tested this by escaping my path on the ENV variable by using ```HTTP_PATH=\/web``` and it parsed OK. 

Hopefully this info helps.

Thanks for your help, man!

**Comment by bkraul** [comment]
I have even better news...based on what I found, it seems that simply doing that escaping before applying the ```TARGET_SERVER``` value should achieve the same result without having to add an extra ```HTTP_PATH``` variable. So basically, I would enter:

```- TARGET_SERVER=10.10.10.130:32400/web```.

Right now, this parses OK when entered as ```- TARGET_SERVER=10.10.10.130:32400\/web```. There seem to be other complications on my end after that which I am trying to resolve.

**Comment by guysoft** [comment]
Ok, if  you have any way to improve the branch, and confirm it works a PR would be great (make sure its on the right branch)

**Comment by bkraul** [comment]
An update for this...I found that the original behavior even in the original master works properly without having to add the ```HTTP_PATH``` env variable. The particular situation with Plex server is that assets are hardcoded to use the ```/web``` directory.

What I ended up doing was to completely override the ```default.conf``` file and add a sub_filter instruction to replace all instances of the ```/web``` subfolder into ```/```, as follows:

```nginx
server {
  listen 80;
  server_name docker;

  location / {
    sub_filter              /web/ /;
    sub_filter_types *;
    sub_filter_once off;
    proxy_pass              http://10.10.10.130:32400/web/;
    proxy_set_header        Host $host;
    proxy_set_header        X-Real-IP $remote_addr;
    proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;

    proxy_connect_timeout   150;
    proxy_send_timeout      100;
    proxy_read_timeout      100;
    proxy_buffers           4 32k;
    client_max_body_size    0;
  }
}
```

This made it work exactly as I was expecting. Needless to say, I believe because this is a specific case and not the rule, there is really no need for an ```HTTP_PATH``` variable, or the target_server branch, as the default behavior works as expected, in case you want to clean that up on your end. Thanks again for your help!

**Comment by bkraul** [comment]
Actually, I will probably submit a PR with the sed reserved character correction I mentioned earlier. Sorry for all the posts, and happy new year.

**Comment by guysoft** [comment]
NP , awaiting PR. At least something is moving now.
Warmly suggest though you don't make commits with no comments and only ... , as in your master now.

**Comment by bkraul** [comment]
I have just issued PR #2 which addresses this issue. I am not sure what you mean about the master. I do not make commits to master. I am making the commits to the ```target_server``` branch and issuing a PR to your ```target_server``` branch.

**Comment by guysoft** [comment]
I meant your master branch had a strange commit in it.
Ok, merging, should I also merge this back in to master? Just making sure you tested this.

**Comment by bkraul** [comment]
I got you. I simply forked your fork and my master is par with your master. I make it a rule to never commit directly to master. I merge from upstream/master into my fork's master instead.

As far as merging back into master, that is up to you. I did test this both for my specific case and also for the normal usage (without a path) and it works correctly. However, you are welcome to test it before applying to master.

I believe with the correction of the sed reserved characters, there can be a little more cleanup (where ```HTTP_PROTOCOL``` might not even be needed anymore, because it can be passed directly through ```TARGET_SERVER```, but that can come later). Also we could try breaking the parsing logic into its own reusable script so as to not crowd ```start.sh```. Sky's the limit, I guess 😄.
